pkg/integration/ttl_integration_test.go

```diff
@@ -0,0 +1,139 @@
+package integration
```

Comment:
I think it would be nice to have an end-to-end example integration test with actual envoy. TTLs are very confusing, so a worked out example with envoy plus a control plane config would help understand it. This doesn't need to happen in this PR since envoy is pegged to some old version in CI, but a follow up would be helpful.